### PR TITLE
573 api integration

### LINKED ISSUE #573: chore(ui): Integrate UI with new LF API
### Describe what should be investigated or refactored
Swap the env vars for the frontend to connect to the new LF API. 

Ensure auth is integrated with the user's session token.

### PR DESCRIPTION
**Pull request by andrewrisse**
closes #573 

Integrates with new LF API and still works with OpenAI.

### CODE REVIEW

File: src/leapfrogai_ui/src/lib/components/AssistantForm.svelte [hunk 1]
```diff
@@ -78,7 +78,7 @@
   });
 
   $: if (data.files) {
-    const fileRows: FileRow[] = data.files.map((file) => ({
+    const fileRows: FileRow[] = data.files?.map((file) => ({
```
Comment (YrrepNoj) [inline src/leapfrogai_ui/src/lib/components/AssistantForm.svelte:81]: Ignorant question;

My understanding of the `?` operator is that it does a 'null/undefined' check against `data.files` before running the mapping function. Is that correct? And if that is correct, isn't that handled by the above `if (data.files)` statement?

File: src/leapfrogai_ui/src/routes/api/chat/+server.ts [hunk 2]
```diff
@@ -28,11 +23,10 @@ export async function POST({ request, locals: { safeGetSession } }) {
   } catch {
     error(400, 'Bad Request');
   }
-
-  // Add the default system prompt to the beginning of the messages
-  if (messages[0].content !== env.DEFAULT_SYSTEM_PROMPT) {
-    messages.unshift({ content: env.DEFAULT_SYSTEM_PROMPT!, role: 'system' });
-  }
+  const openai = createOpenAI({
```
Comment (YrrepNoj) [inline src/leapfrogai_ui/src/routes/api/chat/+server.ts:26]: I've noticed other sections of this PR use `src/lib/server/constants.ts -> getOpenAiClient()`. Does it make sense to use that helper here instead?

**Review by YrrepNoj — DISMISSED** [review]

File: src/leapfrogai_ui/src/lib/components/AssistantForm.svelte (hunk 1 above)
Comment (andrewrisse) [reply src/leapfrogai_ui/src/lib/components/AssistantForm.svelte:81]: it sure does, removing

File: src/leapfrogai_ui/src/routes/api/chat/+server.ts (hunk 2 above)
Comment (andrewrisse) [reply src/leapfrogai_ui/src/routes/api/chat/+server.ts:26]: this is a special case where the vercel ai sdk expects a different version of the openai client so we create it with their helper (for streaming). I will add a comment

**Review by YrrepNoj — APPROVED** [review]

**Pull request merged**